server/auth.js

```diff
@@ -1,137 +1,118 @@
-
-const Boom = require('boom');
+/* eslint-disable consistent-return */
+const { promisify } = require('util');
+const Boom = require('@hapi/boom');
 const jwksRsa = require('jwks-rsa');
 const jwt = require('jsonwebtoken');
-const tools = require('auth0-extension-hapi-tools');
-
 const config = require('../lib/config');
-
-const scopes = [{ value: 'openid' }, { value: 'profile' }];
-
-module.exports.register = (server, options, next) => {
-  const jwtOptions = {
-    dashboardAdmin: {
-      key: config('EXTENSION_SECRET'),
-      verifyOptions: {
-        audience: 'urn:api-account-linking',
-        issuer: config('PUBLIC_WT_URL'),
-        algorithms: ['HS256']
-      }
-    },
-    resourceServer: {
-      key: jwksRsa.hapiJwt2Key({
-        cache: true,
-        rateLimit: true,
-        jwksRequestsPerMinute: 2,
-        jwksUri: `https://${config('AUTH0_DOMAIN')}/.well-known/jwks.json`
-      }),
-      verifyOptions: {
-        audience: 'urn:auth0-account-linking-api',
-        issuer: `https://${config('AUTH0_DOMAIN')}/`,
-        algorithms: ['RS256']
-      }
-    }
-  };
-
-  server.auth.strategy('jwt', 'jwt', {
-    // Get the complete decoded token, because we need info from the header (the kid)
-    complete: true,
-
-    verifyFunc: (decoded, req, callback) => {
-      if (!decoded) {
-        return callback(null, false);
+const plugin = require('../lib/session');
+
+module.exports = {
+  name: 'auth',
+  async register(server) {
+    const jwtOptions = {
+      dashboardAdmin: {
+        key: config('EXTENSION_SECRET'),
+        verifyOptions: {
+          audience: 'urn:api-account-linking',
+          issuer: config('PUBLIC_WT_URL'),
+          algorithms: ['HS256']
+        }
+      },
+      resourceServer: {
+        key: jwksRsa.hapiJwt2KeyAsync({
+          cache: true,
+          rateLimit: true,
+          jwksRequestsPerMinute: 2,
+          jwksUri: `https://${config('AUTH0_DOMAIN')}/.well-known/jwks.json`
+        }),
+        verifyOptions: {
+          audience: 'urn:auth0-account-linking-api',
+          issuer: `https://${config('AUTH0_DOMAIN')}/`,
+          algorithms: ['RS256']
+        }
       }
-
-      const header = req.headers.authorization;
-      if (header && header.indexOf('Bearer ') === 0) {
-        const token = header.split(' ')[1];
-        if (
-          decoded &&
-          decoded.payload &&
-          decoded.payload.iss === `https://${config('AUTH0_DOMAIN')}/`
-        ) {
-          return jwtOptions.resourceServer.key(decoded, (keyErr, key) => {
-            if (keyErr) {
-              return callback(Boom.wrap(keyErr), null, null);
+    };
+
+    server.auth.strategy('jwt', 'jwt', {
+      complete: true,
+      verify: async (decoded, req) => {
+        try {
+          if (!decoded) {
+            return { isValid: false };
+          }
+          const header = req.headers.authorization;
+          if (!header || !header.indexOf('Bearer ') === 0) {
```

Comment:
nit: this reads a bit better to me
```suggestion
          if (!header || !header.startsWith('Bearer ')) {

Reply:
Sounds good, making that change